bridges/centralized-ethereum/src/actors/wit_poller.rs

```diff
@@ -218,64 +281,55 @@ async fn get_dr_timestamp(
+    let protocol_version = if protocol_info
+        .all_versions
+        .get_activation_epoch(protocol_info.current_version)
+        <= dr_last_commit_epoch
+    {
+        protocol_info.current_version
+    } else {
+        protocol_info.current_version.prev()
     };
-    let epoch_constants = EpochConstants {
-        checkpoint_zero_timestamp: get_protocol_version_activation_epoch(V1_7).into(),
-        checkpoints_period: get_protocol_version_period(V1_7),
-        checkpoint_zero_timestamp_wit2: get_protocol_version_activation_epoch(V2_0).into(),
-        checkpoints_period_wit2: get_protocol_version_period(V2_0),
-    };
-    let timestamp = convert_block_epoch_to_timestamp(
-        epoch_constants,
-        block_epoch + u32::from(dr_commits_round + 1),
-    );
+    let protocol_activation_timestamp = protocol_info
+        .derive_activation_timestamp(protocol_version, consensus_constants)
+        .unwrap_or(consensus_constants.checkpoint_zero_timestamp);
+    let protocol_activation_epoch = protocol_info
+        .all_versions
+        .try_get_activation_epoch(protocol_version)
+        .unwrap_or_default();
+    let protocol_checkpoint_period = protocol_info
+        .all_checkpoints_periods
+        .get(&protocol_version)
+        .unwrap_or(&45u16);
 
-    Ok(timestamp)
+    Ok(protocol_activation_timestamp
+        + i64::from((dr_last_commit_epoch - protocol_activation_epoch) * u32::from(*protocol_checkpoint_period)))
 }
 
-async fn get_consensus_constants(
-    witnet_client: Addr<JsonRpcClient>,
-) -> Result<ConsensusConstants, ()> {
-    let method = String::from("getConsensusConstants");
+/// Get current protocol info from the Witnet node
+async fn get_protocol_info(witnet_client: Addr<JsonRpcClient>) -> Result<ProtocolInfo, ()> {
+    let method = String::from("protocol");
     let params = json!(null);
     let req = jsonrpc::Request::method(method)
         .timeout(Duration::from_millis(5_000))
         .params(params)
         .expect("params failed serialization");
-    let result = witnet_client.send(req).await;
-    let result = match result {
-        Ok(result) => result,
+    let report = witnet_client.send(req).await;
+    let report = match report {
+        Ok(report) => report,
         Err(_) => {
             log::error!("Failed to connect to witnet client, will retry later");
             return Err(());
         }
     };
-    let consensus_constants = match result {
-        Ok(value) => serde_json::from_value::<ConsensusConstants>(value)
-            .expect("failed to deserialize consensus constants"),
+    match report {
+        Ok(value) => Ok(serde_json::from_value::<ProtocolInfo>(value)
+            .expect("failed to deserialize protocol info")),
         Err(e) => {
-            log::error!("error in getConsensusConstants call: {:?}", e);
-            return Err(());
+            log::error!("Error when getting protocol info: {:?}", e);
+            Err(())
```

Comment:
```suggestion
         
            Err(())
```

Reply:
All these "returns" are on purpose, as the pretend to exit the function, not the match block.